R/check_soft_duplicates.R

```diff
@@ -0,0 +1,100 @@
+#' Checks for survey similarities - Soft Duplicates
+#'
+#' @param dataset RawData to be checked as a dataframe or a list with dataframe
+#'  stored as "checked_dataset"
+#' @param kobo_survey Kobo Survey Sheet
+#' @param uuid String character for the uuid column. By default "uuid"
+#' @param idnk_value String character for the value of the "I don't know" value
+#' @param sm_seperator Separator for choice multiple questions. The default is "."
+#' @param log_name Name of the log dataframe flagged in the end of the function
+#'
+#' @return return a list with the dataset checked stored as checked_dataset and
+#' a dataframe with the soft duplicate log
+#' @export
+#'
+#' @examples
+#' soft_duplicates <- check_soft_duplicates(dataset = cleaningtools_raw_data,
+#'                                          kobo_survey = cleaningtools_survey,
+#'                                          uuid = "X_uuid",
+#'                                          idnk_value = "dont_know",
+#'                                          sm_seperator = ".",
+#'                                          log_name = "soft_duplicate_log")
+
```

Comment:
Could we add maybe an example to do it per enumerator? 

group_by_enum_raw_data <- cleaningtools_raw_data %>% 
  group_by(enumerator_num) 

soft_per_enum <- group_by_enum_raw_data %>% 
  group_split() %>%
  purrr::map(~check_soft_duplicates(dataset = .,
                                    kobo_survey = cleaningtools_survey,
                                    uuid = "X_uuid",
                                    idnk_value = "dont_know",
                                    sm_seperator = ".",
                                    log_name = "soft_duplicate_log"))

soft_per_enum %>% 
  purrr::map(~.[["soft_duplicate_log"]]) %>%
  purrr::map2(.y = dplyr::group_keys(group_by_enum_raw_data) %>% unlist(), 
              ~mutate(.x, enum = .y)) %>% 
  do.call(bind_rows,.)